core/sup_tree_core/version_upgrade_task_queue.ex

```diff
@@ -14,36 +14,50 @@ defmodule AntikytheraCore.VersionUpgradeTaskQueue do
   use GenServer
   alias Antikythera.GearName
   alias AntikytheraCore.Version
+  require AntikytheraCore.Logger, as: L
 
   def start_link() do
     GenServer.start_link(__MODULE__, :ok, [name: __MODULE__])
   end
 
   @impl true
   def init(:ok) do
-    {:ok, %{queue: :queue.new(), task_pid: nil}}
+    {:ok, %{queue: :queue.new(), task_pid: nil, enabled?: true}}
   end
 
   @impl true
   def handle_cast(message, state) do
-    new_state = enqueue_message(state, message) |> run_task_if_possible()
+    new_state =
+      case message do
+        {:upgrade    , instruction} -> enqueue_message(state, instruction) |> run_task_if_possible()
+        {:set_enabled, enabled?   } -> set_enabled(state, enabled?)
+      end
     {:noreply, new_state}
   end
 
   defp enqueue_message(%{queue: q} = state, message) do
     %{state | queue: :queue.in(message, q)}
   end
 
+  defp set_enabled(state, enabled?) do
+    case {state.enabled?, enabled?} do
+      {true , false} -> L.info("upgrade processing is turned off")
+      {false, true } -> L.info("upgrade processing is turned on")
+      _              -> :ok
+    end
+    %{state | enabled?: enabled?}
+  end
+
   @impl true
   def handle_info({:DOWN, _monitor_ref, :process, pid, _reason}, %{task_pid: pid} = state) do
     new_state = Map.put(state, :task_pid, nil) |> run_task_if_possible()
     {:noreply, new_state}
   end
 
-  defp run_task_if_possible(%{queue: q, task_pid: pid} = state) do
-    if pid == nil do
+  defp run_task_if_possible(%{queue: q, task_pid: pid, enable_upgrade?: enable?} = state) do
```

Comment:
`enable_upgrade?` here should also be renamed.

Reply:
Sorry, I'd overlooked.
fixed in e29d372